doc/contributing.md

```diff
@@ -19,14 +12,225 @@ To make a code contribution to the project:
+      const { value } = action.data;
+      return value;
+    }
+    case FETCH_RANDOM_STRING_FAIL: {
+      return "null";
+    }
+    default:
+      return state;
+  }
+}
+```
+
+And we'll render our new `RandomString` component at the bottom of our profile page. The `Profile` component is defined in `static/js/components/Profile.jsx`:
+
+``` javascript
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import NewTokenForm from './NewTokenForm';
+import TokenList from './TokenList';
+import RandomString from './RandomString';
+
+
+const Profile = (props) => {
+  const profile = useSelector((state) => state.profile);
+  const groups = useSelector((state) => state.groups.latest);
+  return (
+    <div>
+      <div>
+        Username:
+        {profile.username}
+      </div>
+      <br />
+      <div>
+        User roles:
+        {profile.roles}
+      </div>
+      <br />
+      <NewTokenForm
+        profile={profile}
+        groups={groups}
+      />
+      <br />
+      <TokenList tokens={profile.tokens} />
+      <br />
+      <br />
+      <RandomString />
+    </div>
+  );
+};
+
+export default Profile;
+```
+Note that we've only added a single line (`<RandomString />`), plus a few line breaks, to render our new component within the `Profile` component.
+
+Then we add two lines to our main reducer file to import our new reducer from `randomString.js` and add it to our root reducer (`static/js/reducers.js`):
+
+``` javascript
+import { combineReducers } from 'redux';
+
+import { reducer as notificationsReducer } from 'baselayer/components/Notifications';
+
+import sourcesReducer from './ducks/sources';
+import sourceReducer from './ducks/source';
+import plotsReducer from './ducks/plots';
+import groupReducer from './ducks/group';
+import groupsReducer from './ducks/groups';
+import profileReducer from './ducks/profile';
+import usersReducer from './ducks/users';
+import sysInfoReducer from './ducks/sysInfo';
+import dbInfoReducer from './ducks/dbInfo';
+import logoReducer from './ducks/logo';
+import randomStringReducer from './ducks/randomString';
+
+
+const root = combineReducers({
+  source: sourceReducer,
+  sources: sourcesReducer,
+  group: groupReducer,
+  groups: groupsReducer,
+  notifications: notificationsReducer,
+  profile: profileReducer,
+  plots: plotsReducer,
+  logo: logoReducer,
+  users: usersReducer,
+  sysInfo: sysInfoReducer,
+  dbInfo: dbInfoReducer,
+  randomString: randomStringReducer
+});
+
+export default root;
+```
+
+
+#### Back-end
+
+Now let's take a look at what we'll need to add to the back-end to make our new feature work.
+
+We'll start by defining a Tornado handler in a new file located in `skyportal/handlers/api/random_string.py`:
+
+``` python
+import uuid
+from baselayer.app.access import permissions, auth_or_token
+from ..base import BaseHandler
+
+
+class RandomStringHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
```

Comment:
Maybe this docstring is too much to show initially.  I'd add a separate section for that later on.